Tradução:  preface-2e.qmd (#40)
cienciadedatos/pt-r4ds: preface-2e.qmd

# Prefácio da segunda edição {.unnumbered}

Boas vindas à segunda edição do livro "R para Ciência de Dados"!

> Esta é uma revisão importante da primeira edição, removendo material que não consideramos mais útil, adicionando material que desejamos incluir na primeira edição e atualizando o texto e o código de acordo com as melhores práticas.

Reviewer: ```suggestion
Esta é uma revisão importante da primeira edição, removendo material que não consideramos mais útil, adicionando material que gostaríamos de ter incluído na primeira edição e atualizando o texto e o código de acordo com as melhores práticas.
```

> Também estamos muito animados em dar as boas-vindas a uma nova co-autora: Mine Çetinkaya-Rundel, uma renomada educadora em ciência de dados e uma de nossas colegas na Posit (antiga RStudio).

Author: ```suggestion
Também estamos muito animados em dar as boas-vindas a uma nova co-autora: Mine Çetinkaya-Rundel, uma renomada educadora em ciência de dados e uma de nossas colegas na Posit (conhecida anteriormente como RStudio).
```

> Segue abaixo um breve resumo das maiores mudanças:

Author: ```suggestion
Segue abaixo um breve resumo das maiores mudanças nesta edição:
```

> -   A primeira parte do livro foi renomeada para "Jogo completo".

Author: ```suggestion
-   A primeira parte do livro foi renomeada para "Visão geral".
```

> -   A primeira parte do livro foi renomeada para "Jogo completo".
>     O objetivo desta seção é fornecer a você os detalhes aproximados do "jogo completo" da ciência de dados antes de nos aprofundarmos nos detalhes.

Reviewer 2: ```suggestion
-   A primeira parte do livro foi renomeada para "As regras do jogo".
    O objetivo desta seção é fornecer a você os detalhes aproximados das "As regras do jogo" da ciência de dados antes de nos aprofundarmos nos detalhes.
```

Reviewer 2: Acho que "As regras do jogo" pode ser uma boa tradução para whole game (visão geral) porque deixa subentendido que os conceitos apresentados no capitulo serão seguidos durante todo o livro e são de fato o workflow que geralmente é usado em ciência de dados com R.

Author: Oi @nextmarte !
Conversei com a Riva ontem e uma das coisas que citei foi o nome do capítulo "whole game"
Acho que a tradução "jogo completo" fica muito ruim hahaha
"As regras do jogo" eu acho que ainda não dá a mesma ideia

A Riva sugeriu algo como "Panorama geral" (pq a ideia do capítulo é mostrar um exemplo completo).

O que acha?

Author: Aliás, vou jogar a discussão no slack

Author: Mandei no slack, assim mais pessoas podem dar ideias nisso

Reviewer 2: > Oi @nextmarte ! Conversei com a Riva ontem e uma das coisas que citei foi o nome do capítulo "whole game" Acho que a tradução "jogo completo" fica muito ruim hahaha "As regras do jogo" eu acho que ainda não dá a mesma ideia
> 
> A Riva sugeriu algo como "Panorama geral" (pq a ideia do capítulo é mostrar um exemplo completo).
> 
> O que acha?

Acho bom também, eu pensei em visão geral, coloquei no comentário, mas quando fiz a revisão achei que valia considerar "regras do jogo"

>     O objetivo desta seção é fornecer a você os detalhes aproximados do "jogo completo" da ciência de dados antes de nos aprofundarmos nos detalhes.

Author: ```suggestion
    O objetivo desta seção é fornecer a você os detalhes aproximados da "visão geral" da ciência de dados antes de nos aprofundarmos nos detalhes.
```

-   A segunda parte do livro é chamada "Visualizar".
    Esta parte oferece uma cobertura mais abrangente das ferramentas e práticas recomendadas de visualização de dados em comparação à primeira edição.

>     O melhor lugar para obter todos os detalhes ainda é o [livro do ggplot2](http://ggplot2-book.org/), mas agora o R4DS aborda mais técnicas importantes.

Reviewer 2: ```suggestion
    O melhor lugar para obter todos os detalhes ainda é o [livro do ggplot2](http://ggplot2-book.org/), mas agora o R4DS aborda de forma mais abrangente as técnicas importantes.
```

-   A terceira parte do livro agora é chamada de "Transformar" e inclui novos capítulos sobre números, vetores lógicos e valores faltantes (*missing values* - *NA*).
    Esses tópicos eram anteriormente parte do capítulo de transformação de dados, mas precisavam de mais espaço para cobrir todos os detalhes.

-   A quarta parte do livro é chamada de "Importar".

>     São novos capítulos que vão além da leitura de arquivos de texto simples para trabalhar com planilhas, importar dados de bancos de dados (*databases*), trabalhar com grandes conjuntos de dados (*big data*), dados retangulares hierárquicos e extrair dados de sites da web (*web scraping*).

Reviewer: Eu acho que `dados retangulares hierárquicos` não é a melhor tradução, já que é no sentido de "tornar dados hierárquicos em dados retangulares". O início do capítulo [Hierarchical data](https://r4ds.hadley.nz/rectangling) explica melhor o propósito.

Author: eu tinha achado confuso inicialmente, mas com a sugestão do @nextmarte realmente faz mais sentido! vou mudar, obrigada!

>     São novos capítulos que vão além da leitura de arquivos de texto simples para trabalhar com planilhas, importar dados de bancos de dados (*databases*), trabalhar com grandes conjuntos de dados (*big data*), dados retangulares hierárquicos e extrair dados de sites da web (*web scraping*).

Author: ```suggestion
    São novos capítulos que vão além da leitura de arquivos de texto simples para trabalhar com planilhas, importar dados de bancos de dados (*databases*), trabalhar com grandes conjuntos de dados (*big data*), converter dados hierárquicos em dados tabulares e extrair dados de sites da web (*web scraping*).
```

> -   A parte "Programar" permanece, mas foi reescrita do zero para se concentrar nas partes mais importantes da escrita de funções e iteração.

Reviewer 2: ```suggestion
-   A parte "Programação funcional" permanece, mas foi reescrita do zero para se concentrar nas partes mais importantes da escrita de funções e iteração.
```

Reviewer 2: Apesar de nem todos os conceitos de programação funcional serem usados nesse capítulo, vários deles são apresentados, por isso acredito que seja uma boa escolha.

Porém, talvez eu esteja bem errado, já que os autores não usaram o termo functional programming no título, já que também falam de conceitos de outros paradigmas, na dúvida acho que podemos manter programar.

Author: Acho que essa parte poderia ser Programar, pq a seção "funcional"  fica mais focado na parte do purrr.

Eu deixei Programar em vez de programação, pois as outras seções são verbos: Visualizar, transformar, importar, etc. 

Faz sentido?

    Agora, a escrita de funções inclui detalhes sobre como criar funções no estilo tidy (lidando com os desafios de avaliação tidy (*tidy evaluation*)), uma vez que isso se tornou muito mais fácil e importante nos últimos anos.
    Adicionamos um novo capítulo sobre funções importantes do *R base* que você provavelmente encontrará em códigos em R por aí.

-   A parte de modelagem foi removida.

>     Nunca tínhamos espaço suficiente para fazer justiça à modelagem e agora existem recursos muito melhores disponíveis.

Reviewer 2: ```suggestion
    Nunca tivemos espaço suficiente para fazer justiça à modelagem e agora existem recursos muito melhores disponíveis.
```

>     Geralmente, recomendamos o uso do pacote [tidymodels](https://www.tidymodels.org/) e a leitura do livro [Tidy Modeling with R](https://www.tmwr.org/) de Max Kuhn e Julia Silge.

Author: ```suggestion
    Geralmente, recomendamos o uso do pacote [tidymodels](https://www.tidymodels.org/) e a leitura do livro [Tidy Modeling with R](https://www.tmwr.org/) escrito por Max Kuhn e Julia Silge.
```
